src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/UnixIPInterfaceProperties.cs

```diff
@@ -17,11 +17,19 @@ internal abstract class UnixIPInterfaceProperties : IPInterfaceProperties
         private readonly string _dnsSuffix;
         private readonly IPAddressCollection _dnsAddresses;
 
-        public UnixIPInterfaceProperties(UnixNetworkInterface uni)
+        public UnixIPInterfaceProperties(UnixNetworkInterface uni, bool globalConfig = false, string dnsSuffix = null, IPAddressCollection dnsAddresses = null)
```

Comment:
Is this "globalConfig" needed?  Or could it just be, e.g.
```C#
_dnsSuffix = dnsSuffix ?? GetDnsSuffix();
_dnsAddresses = dnsAddresses ?? GetDnsAddresses();
```
or does that cause problems?

Reply:
I did not want to try again if global attempt already fails and dnsSuffix or dnsAddresses is is null. I was focusing on Linux but perhaps best longer term fix is to update all other Unix platforms and remove  GetDns*() completely.